filters/include/bloom_filter.hpp

```diff
@@ -0,0 +1,757 @@
+   */
+  bool query(int8_t item) const;
+
+  /**
+   * Queries the filter with the given 64-bit floating point value and returns whether the value
+   * might have been seen previoiusly. The filter's expected Fale Positive Probability
+   * determines the chances of a true result being a false positive. False engatives are
+   * never possible.
+   * @param item The given double.
+   * @return The result from querying the filter with the given item.
+   */
+  bool query(double item) const;
+
+  /**
+   * Queries the filter with the given 32-bit floating point value and returns whether the value
+   * might have been seen previoiusly. The filter's expected Fale Positive Probability
+   * determines the chances of a true result being a false positive. False engatives are
+   * never possible.
+   * @param item The given float.
+   * @return The result from querying the filter with the given item.
+   */
+  bool query(float item) const;
+
+  /**
+   * Queries the filter with the given data array. and returns the result from
+   * Queries the filter with the given 64-bit floating point value and returns whether the value
+   * might have been seen previoiusly. The filter's expected Fale Positive Probability
+   * determines the chances of a true result being a false positive. False engatives are
+   * never possible.
+   * @param data The given array.
+   * @param length_bytes The array length in bytes.
+   * @return The result from querying the filter with the given item.
+   */
+  bool query(const void* data, size_t length_bytes) const;
+
+  // OTHER OPERATIONS
+
+  /**
+   * Unions two Bloom Filters by applying a logical OR. The result will recognized
+   * any values seen by either filter (as well as false positives).
+   * @param other A BloomFilter to union with this one
+   */
+  void union_with(const bloom_filter_alloc& other);
+
+  /**
+   * Intersects two Bloom Filters by applying a logical AND. The result will recognize
+   * only values seen by both filters (as well as false positives).
+   * @param other A Bloom Filter to union with this one
+   */
+  void intersect(const bloom_filter_alloc& other);
+
+  /**
+   * Inverts all the bits of the BloomFilter. Approximately inverts the notion of set-membership.
+   */
+  void invert();
+
+  /**
+   * Helps identify if two Bloom Filters may be unioned or intersected.
+   * @param other A Bloom Filter to check for compatibility with this one
+   * @return True if the filters are compatible, otherwise false
+   */
+   bool is_compatible(const bloom_filter_alloc& other) const;
+
+  /**
+   * @brief Checks if the Bloom Filter is read-only.
+   *
+   * @return True if the filter is read-only, otherwise false.
+   */
+  bool is_read_only() const;
+
+  /**
+   * @brief Returns whether the filter owns its underlying memory
+   * @return True if the filter owns its memory, otherwise false
+   */
+  bool is_memory_owned() const;
+
+  /**
+   * @brief Checks if the Bloom Filter has backing memory.
```

Comment:
not clear to me what does this mean

Reply:
would `is_wrapped()` work better?

Reply:
not sure. this is the first mention of backing, and no explanation. I am still not sure what that backing is having glanced through the whole PR

Reply:
ok, changed to `is_wrapped()` and `get_wrapped_memory()` which should be more clear